src/Illuminate/Database/DatabaseTransactionsManager.php

```diff
@@ -112,13 +112,22 @@ public function callbacksShouldIgnore(DatabaseTransactionRecord $transaction)
     /**
      * Get the transactions that are applicable to callbacks.
      *
-     * @return \Illuminate\Support\Collection
+     * @return \Illuminate\Support\Collection<int, \Illuminate\Database\DatabaseTransactionRecord>
      */
     public function callbackApplicableTransactions()
     {
-        return $this->transactions->reject(function ($transaction) {
-            return $transaction === $this->callbacksShouldIgnore;
-        })->values();
+        return $this->transactions->values();
```

Comment:
We can simplify this to just 
```suggestion
        return $this->transactions;
```

Reply:
I thought about that, but it would return the same collection instance. I figured it would be better to clone it or create a new one instead.